File: dotnet/Multiplayer/Leaderboard/LeaderboardRequest.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Text.Json.Serialization;
using System.Threading.Tasks;

namespace Michitai.Multiplayer.Leaderboard
{
    /// <summary>
    /// Request data for querying the leaderboard with sorting and limit options.
    /// </summary>
    public class LeaderboardRequest
    {
        [JsonInclude]
        private string[] Sort_by { get; set; } = Array.Empty<string>();
        [JsonInclude]
        private int Limit { get; set; }

        /// <summary>
        /// Initializes a new LeaderboardRequest.
        /// </summary>
        /// <param name="sortBy">Array of field names to sort by.</param>
        /// <param name="limit">Maximum number of results to return.</param>
        public LeaderboardRequest(string[] sortBy, int limit)
        {
            this.Sort_by = sortBy;
            this.Limit = limit;
        }
    }
}

File: java/src/main/java/com/michitai/multiplayer/leaderboard/LeaderboardRequest.java
package com.michitai.multiplayer.leaderboard;

import com.fasterxml.jackson.annotation.JsonProperty;

/**
 * Request for querying the leaderboard with sorting and limit options.
 */
public class LeaderboardRequest {
    @JsonProperty("sort_by")
    private String[] sortBy;

    @JsonProperty("limit")
    private int limit;

    public LeaderboardRequest(String[] sortBy, int limit) {
        this.sortBy = sortBy;
        this.limit = limit;
    }

    public String[] getSortBy() {
        return sortBy;
    }

    public void setSortBy(String[] sortBy) {
        this.sortBy = sortBy;
    }

    public int getLimit() {
        return limit;
    }

    public void setLimit(int limit) {
        this.limit = limit;
    }
}
